aws/src/main/java/org/apache/iceberg/aws/AwsProperties.java

```diff
@@ -975,4 +1008,19 @@ private <T extends SdkClientBuilder> void configureEndpoint(T builder, String en
       builder.endpointOverride(URI.create(endpoint));
     }
   }
+
+  @VisibleForTesting
+  <T extends ApacheHttpClient.Builder> void configureApacheHttpClientBuilder(T builder) {
+    boolean setConnectionTimeout = apacheHttpClientConnectionTimeout != null;
+    boolean setSocketTimeout = apacheHttpClientSocketTimeout != null;
+    if (setConnectionTimeout && setSocketTimeout) {
+      builder
```

Comment:
we don't need to do this, when you do `builder.xxxx` it automatically mutates the builder itself. Returning itself is just to facilitate method chaining.

Reply:
Thank you for your review. However, I think I may not get what you mean. Do you refer to line 1017?  This method `configureApacheHttpClientBuilder` does not have return value given its return type is `void`. The `builder` standing alone in line 1017 is just due to the autoformatting. Indeed, I design this method so that it can be used like:
```
ApacheHttpClient.builder().applyMutation(this::configureApacheHttpClientBuilder)
```
like the one in line 923.

If I misunderstood your point, please let me know. Thank you again for your help!